tensorflow/python/kernel_tests/xent_op_deterministic_test.py

```diff
@@ -94,39 +73,132 @@ def _generateInputs(self, dtype, seed=123):
 
   @test_util.run_in_graph_and_eager_modes
   def testForward(self):
-    with self.session(), test_util.force_cpu():
-      for dtype in [np.float16, np.float32, np.float64]:
-        for trial in range(5):
-          seed = 123 + trial
-          labels, logits = self._generateInputs(dtype, seed=seed)
-          result_a = nn_ops.softmax_cross_entropy_with_logits_v2(
-              labels=labels, logits=logits)
-          result_b = nn_ops.softmax_cross_entropy_with_logits_v2(
-              labels=labels, logits=logits)
-          self.assertAllEqual(result_a, result_b)
+    for use_gpu in [False, True]:
+      if use_gpu and self._gpuNotAvailable():
+        continue
+      with self.cached_session(), test_util.device(use_gpu):
+        # Even in eager mode, the above line will be able to pin ops to CPU.
+        for dtype in [np.float16, np.float32, np.float64]:
+          for trial in range(5):
+            seed = 123 + trial
+            labels, logits = self._generateInputs(
+                dtype, seed=seed, forward_not_backward=True)
+            result_a = nn_ops.softmax_cross_entropy_with_logits(
+                labels=labels, logits=logits)
+            result_b = nn_ops.softmax_cross_entropy_with_logits(
+                labels=labels, logits=logits)
+            self.assertAllEqual(result_a, result_b)
+
+  def _testBackward(self, labels_not_logits=False):
+    for use_gpu in [False, True]:
+      if use_gpu and self._gpuNotAvailable():
+        continue
+      with self.cached_session(), test_util.device(use_gpu):
+        # Even in eager mode, the above line will be able to pin ops to CPU.
+        for dtype in [np.float16, np.float32, np.float64]:
+          labels, logits = self._generateInputs(dtype, seed=456)
+          output_shape = labels.shape[0]
+
+          def gradients(seed=789):
```

Comment:
No reason for seed to have a default value here

Reply:
Noted.